.github/workflows/integration-tests.yaml

```diff
@@ -0,0 +1,212 @@
+name: Integration Tests
+on:
+  schedule:
+    # Run nightly at 2 AM UTC
+    - cron: '0 2 * * *'
+  workflow_dispatch:
+    # Allow manual triggering for debugging
+    inputs:
+      cloud:
+        description: 'Cloud provider to test (aws, azure, or both)'
+        required: false
+        default: 'both'
+        type: choice
+        options:
+          - both
+          - aws
+          - azure
+
+jobs:
+  integration-tests:
+    runs-on: 
```

Comment:
FYI: This is temporary, we will need to work on dedicated infra for this since we have integration tests now